### PR TITLE
Use ghc+integer-gmp for alpine linux build release

### PR DESCRIPTION
**Pull request by jneira**
* [After diagnose the relese build was failing](https://github.com/haskell/haskell-language-server/pull/2606#issuecomment-1017227934) for alpine+ghc with integer-simple due to a new package requiring integer-gmp (https://hackage.haskell.org/package/mod)
* Pinged @maerwald and kindly added in ghcup the missinf ghc versions for alpine with integer-gmp (https://github.com/haskell/ghcup-metadata/pull/15)
* So we can remove the existing workaround here and let haskell/action download the new bindists

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2615"></a>

### CODE REVIEW

**Comment by jneira** [comment]
* Checking the release build here: https://github.com/jneira/haskell-language-server/actions/runs/1725135759

**Comment by jneira** [comment]
The build has progressed but it has failed for linux and ghc-9.2.1 with

```
src/Ide/Version.hs:24:21: error:
Error:     • Exception when trying to run compile-time code:
        PATH: getEnv: does not exist (no environment variable)
      Code: gitCommitCount
    • In the untyped splice: $gitCommitCount
   |
24 |   let commitCount = $gitCommitCount
   |                     ^^^^^^^^^^^^^^^
cabal: Failed to build hls-1.5.1.0 (which is required by exe:hls from
hls-1.5.1.0).
```
🤔

**Comment by hasufell** [comment]
> The build has progressed but it has failed for linux and ghc-9.2.1 with
> 
> ```
> src/Ide/Version.hs:24:21: error:
> Error:     • Exception when trying to run compile-time code:
>         PATH: getEnv: does not exist (no environment variable)
>       Code: gitCommitCount
>     • In the untyped splice: $gitCommitCount
>    |
> 24 |   let commitCount = $gitCommitCount
>    |                     ^^^^^^^^^^^^^^^
> cabal: Failed to build hls-1.5.1.0 (which is required by exe:hls from
> hls-1.5.1.0).
> ```
> 
> thinking

Sounds like this part is failing: https://github.com/acfoltzer/gitrev/blob/450a7abf444260e95791649489705e54650c0232/src/Development/GitRev.hs#L65

The environment is borked (doesn't contain the `PATH` env var).

**Comment by jneira** [comment]
it does (https://github.com/jneira/haskell-language-server/runs/4889676070?check_suite_focus=true)

```
Path: /__t/stack/2.7.3/x64:/github/home/.ghcup/bin:/github/home/.ghcup/ghc/9.2.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

the build is successful for all ghcs < 9.2.1, using the same docker image for all, and the build for ghc-9.2.1 is successful in Ubuntu so I would say it is a weird interaction between both

**Comment by jneira** [comment]
well the workflow between the Ubuntu build and the alpine one 8s not exactly the same but I don't see any change which could cause the bug
will try to use Ubuntu in the same workflow to check it though (and as possible workaround)

also we could trace the value of the PATH env bar in our code but getExecutable is pretty standard

**Comment by jneira** [comment]
> the build for ghc-9.2.1 is successful in Ubuntu so I would say it is a weird interaction between both

well the ghc version for alpine is not exactly the same than the Ubuntu one

**Comment by jneira** [comment]
As first sanity check i am gonna invalidate the cache created in ubuntu

**Comment by jneira** [comment]
A fresh cache did not fix it: https://github.com/jneira/haskell-language-server/runs/4893126248?check_suite_focus=true

**Comment by jneira** [comment]
As expected changing alpine for ubuntu for the ghc-9.2.1 has been succesful so the workflow is not the culprit: https://github.com/jneira/haskell-language-server/runs/4893355328?check_suite_focus=true

At this point i think we could consider use ubuntu for ghc-9.2.1 (it is experimental anyways) until we fix the bug for alpine + ghc-9.2.1. That way we will unblock the incoming release.

//cc @hasufell @pepeiborra @michaelpj

**Comment by michaelpj** [comment]
@Bodigrim made a release of `mod` with `integer-simple` (https://github.com/Bodigrim/mod/issues/13#issuecomment-1017944515), so if we bump the index-state we may find that cabal can now get a build plan when using `integer-simple`.

**Comment by michaelpj** [comment]
Personally, the commit count suffix on the version string doesn't seem that useful to me, we could consider just binning that.

**Comment by hasufell** [comment]
> @Bodigrim made a release of `mod` with `integer-simple` (https://github.com/Bodigrim/mod/issues/13#issuecomment-1017944515), so if we bump the index-state we may find that cabal can now get a build plan when using `integer-simple`.

Probably won't, because I already updated ghcup alpine bindists to integer-gmp.

**Comment by pepeiborra** [comment]
I support dropping the dependency if it helps.

**Comment by jneira** [comment]
All `gitrev` definitions needs find git in the env var PATH throught `findExecutable "git"` . We are actually using it to get the git commit here:

https://github.com/haskell/haskell-language-server/blob/b937baa03cf310109aafc0d4835cdff1240ffd3e/src/Ide/Arguments.hs#L141-L150

and i would like to keep the commit when doing `hls --version`

The alternative `githash` dont use `findExecutable` though but does a raw `(proc "git" args) { cwd = Just root }` so maybe it would work :shrug:

**Comment by jneira** [comment]
Meanwhile i've reproduced the ghc bug in a fresh alpine-3.12 vm installing ghc with ghcup:

imagen

wonderful

**Comment by jneira** [comment]
It worths to note that ghc-9.2.1 in alpine seems to be fragile:
* "Alpine segfaults many tests in 9.2.1" https://gitlab.haskell.org/ghc/ghc/-/issues/20266
* "Alpine ci job is broken" https://gitlab.haskell.org/ghc/ghc/-/issues/20574

**Comment by jneira** [comment]
Trying githash in alpine+9.2.1 here: https://github.com/jneira/haskell-language-server/runs/4896130017?check_suite_focus=true

**Comment by jneira** [comment]
Ok, githash worked, yay! thanks @hasufell for the suggestion
I stand a little bit disturbed by the bug but whatever

**Comment by hasufell** [comment]
@jneira I've updated the ghc 9.2.1 alpine bindist in ghcup. `getEnv` should work now.

**Comment by jneira** [comment]
@hasufell many thanks! i've reran the build workflow and it should use the new ghc bindist: https://github.com/haskell/haskell-language-server/actions/runs/1728847536

**Comment by jneira** [comment]
> @jneira I've updated the ghc 9.2.1 alpine bindist in ghcup. `getEnv` should work now.

i've double checked in my vm and it works as intended 👍

**Comment by jneira** [comment]
Well i think we can keep githash cause as commented by @hasufell it is being actively maintained so the pr could be merged if you agree

**Review by pepeiborra — APPROVED** [review]
The githash change looks fine to me, but I haven't reviewed the CI changes. Maybe @Anton-Latukha can take a look.

**Review by michaelpj — APPROVED** [review]

**Comment by Anton-Latukha** [comment]
Well, as I came around to look at it... To review it, I needed/need to fully learn & understand the `build` workflow, so time.

Seems good. Would submit PRs if would find something.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -97,36 +84,36 @@ jobs:
...
       uses: actions/upload-artifact@v2
       with:
-        name: haskell-language-server-${{ runner.OS }}-${{ matrix.ghc }}${{env.EXE_EXT}}.${{ steps.compress_server_binary.outputs.extension }}
+        name: haskell-language-server-${{ runner.os }}-${{ matrix.ghc }}${{ env.EXE_EXT }}.${{ steps.compress_server_binary.outputs.extension }}
         path: ${{ steps.compress_server_binary.outputs.path }}
 
-    - name: (GHC 8.10) Build the wrapper
+    - name: Build the wrapper
       if: matrix.ghc == '8.10.7'
       run: cabal build exe:hls-wrapper -O2 $LINUX_CABAL_ARGS
```
Comment (pepeiborra) [inline .github/workflows/build.yml:106]: does this need to be `$CABAL_ARGS` ?

Comment (jneira) [reply .github/workflows/build.yml:106]: omg 😧

Comment (jneira) [reply .github/workflows/build.yml:106]: that should have failed, as @hasufell suggested, setting the appropriate bash flag

**Pull request merged**